src/Templates/ModelTemplate.cshtml

```diff
@@ -426,51 +426,41 @@ else
     var resultVar = ftg.ResultTypeName.ToShortName();
     var resultVarTarget = resultVar;
     var futureTypeName = $"{Model.CodeModel.Namespace}.{Model.Name}";
-    if (ftg.ResultType is IteratorTypeGo)
+    if (ftg.ResultType is IteratorTypeGo itg)
     {
-        resultVarTarget = $"{resultVarTarget}.{ftg.ResultType.Cast<IteratorTypeGo>().PageField}";
+        resultVarTarget = $"{resultVarTarget}.{itg.PageField}";
     }
     <text>
     // Result returns the result of the asynchronous operation.
     // If the operation has not completed it will return an error.
-    func (future @Model.Name) Result(client @ftg.ClientTypeName) (@resultVar @ftg.ResultTypeName, err error) {
+    func (future *@Model.Name) Result(client @ftg.ClientTypeName) (@resultVar @ftg.ResultTypeName, err error) {
     var done bool
     done, err = future.Done(client)
     if err != nil {
         err = autorest.NewErrorWithError(err, "@futureTypeName", "Result", future.Response(), "Polling failure")
         return
     }
     if !done {
-        return @resultVar, azure.NewAsyncOpIncompleteError("@futureTypeName")
-    }
-    if future.PollingMethod() == azure.PollingLocation {
-        @resultVarTarget, err = client.@(ftg.ResponderMethodName)(future.Response())
-        if err != nil {
-            err = autorest.NewErrorWithError(err, "@futureTypeName", "Result", future.Response(), "Failure responding to request")
-        }
+        err = azure.NewAsyncOpIncompleteError("@futureTypeName")
         return
     }
-    var req *http.Request
-    var resp *http.Response
-    if future.PollingURL() != "" {
-        req, err = http.NewRequest(http.MethodGet, future.PollingURL(), nil)
+    @if (ftg.ResultTypeName == MethodGo.DefaultReturnType)
+    {
+    @:@(resultVar).Response = future.Response()
+    }
+    else
```

Comment:
is this for that case where you need to do an extra get at the end ? if so, what is `ResultTypeName` then ?

Reply:
The default return type is ```autorest.Response``` so if that's all that's being returned then we assign the raw HTTP response to the ```Response``` field.

Reply:
Yes, I guess the only think im saying is that maybe its not obvious why if the response is not the default return type we need to have the extra getResult on the future.

 Nit: Maybe a comment or a helper method.

---
In reply to: [191599956](https://github.com/Azure/autorest.go/pull/129#discussion_r191599956) [](ancestors = 191599956)

Reply:
Added a comment.